### PR TITLE
librbd: add override in header files

### PR DESCRIPTION
**Pull request by Liuchang0812**
Fixes: http://tracker.ceph.com/issues/19012

Signed-off-by: liuchang0812 <liuchang0812@gmail.com>

### CODE REVIEW

**Comment by Liuchang0812** [comment]
@dillaman @liewegas Please take a look.   as we talked in #13437

File: src/librbd/AioImageRequest.h [hunk 1]
```diff
@@ -172,30 +172,30 @@ class AioImageWrite : public AbstractAioImageWrite<ImageCtxT> {
...
 
   void assemble_extent(const ObjectExtent &object_extent, bufferlist *bl);
 
-  virtual void send_image_cache_request() override;
+  void send_image_cache_request() override;
 
-  virtual void send_object_cache_requests(const ObjectExtents &object_extents,
-                                          uint64_t journal_tid);
+  void send_object_cache_requests(const ObjectExtents &object_extents,
+                                          uint64_t journal_tid) override;
```
Comment (dillaman) [inline src/librbd/AioImageRequest.h]: Nit: indentation

File: src/librbd/AioImageRequest.h [hunk 2]
```diff
@@ -172,30 +172,30 @@ class AioImageWrite : public AbstractAioImageWrite<ImageCtxT> {
...
 
-  virtual void send_object_requests(const ObjectExtents &object_extents,
+  void send_object_requests(const ObjectExtents &object_extents,
                                     const ::SnapContext &snapc,
```
Comment (dillaman) [inline src/librbd/AioImageRequest.h]: Nit: indentation

File: src/librbd/AioImageRequest.h [hunk 3]
```diff
@@ -172,30 +172,30 @@ class AioImageWrite : public AbstractAioImageWrite<ImageCtxT> {
...
+  AioObjectRequestHandle *create_object_request(
       const ObjectExtent &object_extent, const ::SnapContext &snapc,
-      Context *on_finish);
+      Context *on_finish) override;
 
-  virtual uint64_t append_journal_event(const AioObjectRequests &requests,
-                                        bool synchronous);
-  virtual void update_stats(size_t length);
+  uint64_t append_journal_event(const AioObjectRequests &requests,
+                                        bool synchronous) override;
```
Comment (dillaman) [inline src/librbd/AioImageRequest.h]: Nit: indentation

File: src/librbd/AioImageRequest.h [hunk 4]
```diff
@@ -213,28 +213,28 @@ class AioImageDiscard : public AbstractAioImageWrite<ImageCtxT> {
...
 
-  virtual void send_image_cache_request() override;
+  void send_image_cache_request() override;
 
-  virtual uint32_t get_object_cache_request_count(bool journaling) const override;
-  virtual void send_object_cache_requests(const ObjectExtents &object_extents,
-                                          uint64_t journal_tid);
+  uint32_t get_object_cache_request_count(bool journaling) const override;
+  void send_object_cache_requests(const ObjectExtents &object_extents,
+                                          uint64_t journal_tid) override;
```
Comment (dillaman) [inline src/librbd/AioImageRequest.h]: Nit: indentation

File: src/librbd/AioImageRequest.h [hunk 5]
```diff
@@ -213,28 +213,28 @@ class AioImageDiscard : public AbstractAioImageWrite<ImageCtxT> {
...
+  AioObjectRequestHandle *create_object_request(
       const ObjectExtent &object_extent, const ::SnapContext &snapc,
-      Context *on_finish);
+      Context *on_finish) override;
 
-  virtual uint64_t append_journal_event(const AioObjectRequests &requests,
-                                        bool synchronous);
-  virtual void update_stats(size_t length);
+  uint64_t append_journal_event(const AioObjectRequests &requests,
+                                        bool synchronous) override;
```
Comment (dillaman) [inline src/librbd/AioImageRequest.h]: Nit: indentation

File: src/librbd/ExclusiveLock.h [hunk 6]
```diff
@@ -29,13 +29,13 @@ class ExclusiveLock : public ManagedLock<ImageCtxT> {
...
-  virtual void post_acquire_lock_handler(int r, Context *on_finish);
-  virtual void pre_release_lock_handler(bool shutting_down,
-                                        Context *on_finish);
-  virtual void post_release_lock_handler(bool shutting_down, int r,
-                                         Context *on_finish);
+  void shutdown_handler(int r, Context *on_finish) override;
+  void pre_acquire_lock_handler(Context *on_finish) override;
+  void post_acquire_lock_handler(int r, Context *on_finish) override;
+  void pre_release_lock_handler(bool shutting_down,
+                                        Context *on_finish) override;
```
Comment (dillaman) [inline src/librbd/ExclusiveLock.h]: Nit: indentation

File: src/librbd/ExclusiveLock.h [hunk 7]
```diff
@@ -29,13 +29,13 @@ class ExclusiveLock : public ManagedLock<ImageCtxT> {
...
+  void post_release_lock_handler(bool shutting_down, int r,
+                                         Context *on_finish) override;
```
Comment (dillaman) [inline src/librbd/ExclusiveLock.h]: Nit: indentation

File: src/librbd/ImageWatcher.h [hunk 8]
```diff
@@ -245,10 +245,10 @@ class ImageWatcher : public Watcher {
   void process_payload(uint64_t notify_id, uint64_t handle,
                              const watch_notify::Payload &payload, int r);
 
-  virtual void handle_notify(uint64_t notify_id, uint64_t handle,
-                             uint64_t notifier_id, bufferlist &bl);
-  virtual void handle_error(uint64_t cookie, int err);
-  virtual void handle_rewatch_complete(int r);
+  void handle_notify(uint64_t notify_id, uint64_t handle,
+                             uint64_t notifier_id, bufferlist &bl) override;
```
Comment (dillaman) [inline src/librbd/ImageWatcher.h]: Nit: indentation

File: src/librbd/LibrbdWriteback.h [hunk 9]
```diff
@@ -21,28 +21,28 @@ namespace librbd {
     LibrbdWriteback(ImageCtx *ictx, Mutex& lock);
 
     // Note that oloc, trunc_size, and trunc_seq are ignored
-    virtual void read(const object_t& oid, uint64_t object_no,
+    void read(const object_t& oid, uint64_t object_no,
 		      const object_locator_t& oloc, uint64_t off, uint64_t len,
```
Comment (dillaman) [inline src/librbd/LibrbdWriteback.h]: Nit: indentation

File: src/librbd/LibrbdWriteback.h [hunk 10]
```diff
@@ -21,28 +21,28 @@ namespace librbd {
...
 		      snapid_t snapid, bufferlist *pbl, uint64_t trunc_size,
-		      __u32 trunc_seq, int op_flags, Context *onfinish);
+		      __u32 trunc_seq, int op_flags, Context *onfinish) override;
 
     // Determine whether a read to this extent could be affected by a
     // write-triggered copy-on-write
-    virtual bool may_copy_on_write(const object_t& oid, uint64_t read_off,
-				   uint64_t read_len, snapid_t snapid);
+    bool may_copy_on_write(const object_t& oid, uint64_t read_off,
+				   uint64_t read_len, snapid_t snapid) override;
```
Comment (dillaman) [inline src/librbd/LibrbdWriteback.h]: Nit: indentation

File: src/librbd/LibrbdWriteback.h [hunk 11]
```diff
@@ -21,28 +21,28 @@ namespace librbd {
...
 
     // Note that oloc, trunc_size, and trunc_seq are ignored
-    virtual ceph_tid_t write(const object_t& oid, const object_locator_t& oloc,
+    ceph_tid_t write(const object_t& oid, const object_locator_t& oloc,
 			     uint64_t off, uint64_t len,
```
Comment (dillaman) [inline src/librbd/LibrbdWriteback.h]: Nit: indentation

File: src/librbd/MirroringWatcher.h [hunk 12]
```diff
@@ -58,8 +58,8 @@ class MirroringWatcher : public Watcher {
   bool handle_payload(const mirroring_watcher::UnknownPayload &payload,
                       Context *on_notify_ack);
 
-  virtual void handle_notify(uint64_t notify_id, uint64_t handle,
-                             uint64_t notifier_id, bufferlist &bl);
+  void handle_notify(uint64_t notify_id, uint64_t handle,
+                             uint64_t notifier_id, bufferlist &bl) override;
```
Comment (dillaman) [inline src/librbd/MirroringWatcher.h]: Nit: indentation

File: src/librbd/Watcher.h [hunk 13]
```diff
@@ -114,11 +114,11 @@ class Watcher {
 
     WatchCtx(Watcher &parent) : watcher(parent) {}
 
-    virtual void handle_notify(uint64_t notify_id,
+    void handle_notify(uint64_t notify_id,
                                uint64_t handle,
```
Comment (dillaman) [inline src/librbd/Watcher.h]: Nit: indentation

File: src/librbd/cache/PassthroughImageCache.h [hunk 14]
```diff
@@ -22,20 +22,20 @@ class PassthroughImageCache : public ImageCache {
...
   /// client AIO methods
-  virtual void aio_read(Extents&& image_extents, ceph::bufferlist *bl,
-                        int fadvise_flags, Context *on_finish);
-  virtual void aio_write(Extents&& image_extents, ceph::bufferlist&& bl,
-                         int fadvise_flags, Context *on_finish);
-  virtual void aio_discard(uint64_t offset, uint64_t length,
-                           Context *on_finish);
-  virtual void aio_flush(Context *on_finish);
+  void aio_read(Extents&& image_extents, ceph::bufferlist *bl,
+                        int fadvise_flags, Context *on_finish) override;
```
Comment (dillaman) [inline src/librbd/cache/PassthroughImageCache.h]: Nit: indentation

File: src/librbd/cache/PassthroughImageCache.h [hunk 15]
```diff
@@ -22,20 +22,20 @@ class PassthroughImageCache : public ImageCache {
...
+  void aio_write(Extents&& image_extents, ceph::bufferlist&& bl,
+                         int fadvise_flags, Context *on_finish) override;
```
Comment (dillaman) [inline src/librbd/cache/PassthroughImageCache.h]: Nit: indentation

File: src/librbd/cache/PassthroughImageCache.h [hunk 16]
```diff
@@ -22,20 +22,20 @@ class PassthroughImageCache : public ImageCache {
...
+  void aio_discard(uint64_t offset, uint64_t length,
+                           Context *on_finish) override;
```
Comment (dillaman) [inline src/librbd/cache/PassthroughImageCache.h]: Nit: indentation

File: src/tools/rbd_mirror/MirrorStatusWatcher.h [hunk 17]
```diff
@@ -17,8 +17,8 @@ class MirrorStatusWatcher : protected librbd::Watcher {
   void shut_down(Context *on_finish);
 
 protected:
-  virtual void handle_notify(uint64_t notify_id, uint64_t handle,
-			     uint64_t notifier_id, bufferlist &bl);
+  void handle_notify(uint64_t notify_id, uint64_t handle,
+			     uint64_t notifier_id, bufferlist &bl) override;
```
Comment (dillaman) [inline src/tools/rbd_mirror/MirrorStatusWatcher.h]: Nit: indentation

**Comment by Liuchang0812** [comment]
@dillaman what's wrong with indentation

**Comment by dillaman** [comment]
@Liuchang0812 The params on the next line should align with the params on the previous line -- they don't now after the "virtual " keyword was removed.

**Comment by Liuchang0812** [comment]
@dillaman thanks I don't know how to fix this by clang-tidy automatically , I will correct it by hands tomorrow

File: src/librbd/LibrbdWriteback.h (hunk 9 above)
Comment (Liuchang0812) [inline src/librbd/LibrbdWriteback.h]: done

File: src/librbd/Watcher.h (hunk 13 above)
Comment (Liuchang0812) [inline src/librbd/Watcher.h]: done

File: src/librbd/cache/PassthroughImageCache.h (hunk 14 above)
Comment (Liuchang0812) [inline src/librbd/cache/PassthroughImageCache.h]: done

File: src/librbd/AioImageRequest.h (hunk 1 above)
Comment (Liuchang0812) [inline src/librbd/AioImageRequest.h]: done

File: src/librbd/AioImageRequest.h (hunk 2 above)
Comment (Liuchang0812) [inline src/librbd/AioImageRequest.h]: done

File: src/librbd/AioImageRequest.h (hunk 3 above)
Comment (Liuchang0812) [inline src/librbd/AioImageRequest.h]: done

File: src/librbd/AioImageRequest.h (hunk 4 above)
Comment (Liuchang0812) [inline src/librbd/AioImageRequest.h]: done

File: src/librbd/AioImageRequest.h (hunk 5 above)
Comment (Liuchang0812) [inline src/librbd/AioImageRequest.h]: done

File: src/librbd/ExclusiveLock.h (hunk 6 above)
Comment (Liuchang0812) [inline src/librbd/ExclusiveLock.h]: done

File: src/librbd/ExclusiveLock.h (hunk 7 above)
Comment (Liuchang0812) [inline src/librbd/ExclusiveLock.h]: done

File: src/librbd/ImageWatcher.h (hunk 8 above)
Comment (Liuchang0812) [inline src/librbd/ImageWatcher.h]: done

File: src/librbd/LibrbdWriteback.h (hunk 10 above)
Comment (Liuchang0812) [inline src/librbd/LibrbdWriteback.h]: done

File: src/librbd/LibrbdWriteback.h (hunk 11 above)
Comment (Liuchang0812) [inline src/librbd/LibrbdWriteback.h]: done

File: src/librbd/MirroringWatcher.h (hunk 12 above)
Comment (Liuchang0812) [inline src/librbd/MirroringWatcher.h]: done

File: src/librbd/cache/PassthroughImageCache.h (hunk 15 above)
Comment (Liuchang0812) [inline src/librbd/cache/PassthroughImageCache.h]: done

File: src/librbd/cache/PassthroughImageCache.h (hunk 16 above)
Comment (Liuchang0812) [inline src/librbd/cache/PassthroughImageCache.h]: done

File: src/tools/rbd_mirror/MirrorStatusWatcher.h (hunk 17 above)
Comment (Liuchang0812) [inline src/tools/rbd_mirror/MirrorStatusWatcher.h]: done

File: src/tools/rbd_mirror/LeaderWatcher.h [hunk 18]
```diff
@@ -104,20 +104,20 @@ class LeaderWatcher : protected librbd::Watcher {
...
         // lock is owned at this point
         Mutex::Locker locker(Parent::m_lock);
         Parent::set_state_post_acquiring();
       }
       watcher->handle_post_acquire_leader_lock(r, on_finish);
     }
-    virtual void pre_release_lock_handler(bool shutting_down,
-                                          Context *on_finish) {
+    void pre_release_lock_handler(bool shutting_down,
+                                          Context *on_finish) override {
```
Comment (Liuchang0812) [inline src/tools/rbd_mirror/LeaderWatcher.h]: done

**Comment by Liuchang0812** [comment]
@dillaman It seems that I have fixed all indentation problems. Could you take a look again. Very thanks.

**Comment by Liuchang0812** [comment]
There are some CE because I rebased branch. I'm fixing it.

**Pull request merged**